Userland/Libraries/LibJS/Runtime/ArrayBuffer.h

```diff
@@ -27,6 +27,12 @@ enum class PreserveResizability {
     PreserveResizability
 };
 
+// 6.2.9 Data Blocks, https://tc39.es/ecma262/#sec-data-blocks
+struct DataBlock {
+    Variant<Empty, ByteBuffer, ByteBuffer*> buffer;
+    bool is_shared;
```

Comment:
Let's use an enum for this

```
struct DataBlock {
    Variant<Empty, ByteBuffer, ByteBuffer*> buffer;
     enum class Shared { No, Yes } is_shared = { Shared::No }
}
```

That way the constructor call sites look like

```
return DataBlock { move(buffer), DataBlock::Shared::No };
```
or 
```
return DataBlock { move(buffer), DataBlock::Shared::Yes };
```

And we can avoid the "what does true/false mean here" conundrum.

Reply:
Thanks for your input. I declared the enum class slightly different, otherwise clang-format made it look a bit odd.